relayer/packet-tx.go

```diff
@@ -23,21 +31,52 @@ func (c *Chain) SendTransferMsg(ctx context.Context, log *zap.Logger, dst *Chain
 		return err
 	}
 
-	var (
-		timeoutHeight    uint64
-		timeoutTimestamp uint64
-	)
+	// if the timestamp offset is set we need to query the dst chains consensus state to get the current time
+	var consensusState ibcexported.ConsensusState
+	if toTimeOffset > 0 {
+		clientStateRes, err := dst.ChainProvider.QueryClientStateResponse(ctx, dsth, dst.ClientID())
+		if err != nil {
+			return fmt.Errorf("failed to query the client state response: %w", err)
+		}
+		clientState, err := clienttypes.UnpackClientState(clientStateRes.ClientState)
+		if err != nil {
+			return fmt.Errorf("failed to unpack client state: %w", err)
+		}
+		consensusStateRes, err := dst.ChainProvider.QueryClientConsensusState(ctx, dsth, dst.ClientID(), clientState.GetLatestHeight())
+		if err != nil {
+			return fmt.Errorf("failed to query client consensus state: %w", err)
+		}
+		consensusState, err = clienttypes.UnpackConsensusState(consensusStateRes.ConsensusState)
+		if err != nil {
+			return fmt.Errorf("failed to unpack consensus state: %w", err)
+		}
+
+		// use local clock time as reference time if it is later than the
+		// consensus state timestamp of the counter party chain, otherwise
+		// still use consensus state timestamp as reference.
+		// see https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/client/cli/tx.go#L94-L110
+		now := time.Now().UnixNano()
+		consensusTimestamp := consensusState.GetTimestamp()
+		if now > 0 {
```

Comment:
Generally it would be preferable to invert the if for an early return, e.g.

```go
if now <= 0 {
  return fmt.Errorf("...")
}
```

so that the other branch of the `if` is less indented and therefore a bit easier to follow.

That being said, I don't think I've ever seen a check for local time being set before 1970. There are so many other things that would likely go wrong on a system claiming to be running with a negative unix epoch, that this check doesn't seem necessary to me.

Reply:
Yeah I thought it was kind of odd as well, I'm just going to yank that check out.